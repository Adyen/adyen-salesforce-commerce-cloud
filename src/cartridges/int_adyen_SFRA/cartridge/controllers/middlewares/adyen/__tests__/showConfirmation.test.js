/* eslint-disable global-require */
let showConfirmation;
let adyenConfigs;
let res;
let req;

beforeEach(() => {
  const { adyen } = require('../../index');
  adyenConfigs = require('*/cartridge/scripts/util/adyenConfigs');
  showConfirmation = adyen.showConfirmation;
  jest.clearAllMocks();

  res = {
    redirect: jest.fn(),
    render: jest.fn(),
  };

  req = {
    querystring: {
      merchantReference: "0",
      signature: 'mocked_signature',
    },
    locale: { id: 'nl_NL' },
  };
});

afterEach(() => {
  jest.resetModules();
  jest.clearAllMocks();
});

describe('Show Confirmation', () => {
  it('should have redirectResult', () => {
    const adyenCheckout = require('*/cartridge/scripts/adyenCheckout');
    const adyenHelper = require('*/cartridge/scripts/util/adyenHelper');
    adyenHelper.getOrderMainPaymentInstrumentType.mockReturnValue('AdyenComponent');
    req.querystring.redirectResult = 'mocked_redirect_result';
    showConfirmation(req, res, jest.fn());
    expect(adyenCheckout.doPaymentsDetailsCall.mock.calls).toMatchSnapshot();
  });

  it('should have payload', () => {
    const adyenCheckout = require('*/cartridge/scripts/adyenCheckout');
    const adyenHelper = require('*/cartridge/scripts/util/adyenHelper');
    adyenHelper.getOrderMainPaymentInstrumentType.mockReturnValue('AdyenComponent');
    req.querystring.payload = 'mocked_payload_result';
    showConfirmation(req, res, jest.fn());
    expect(adyenCheckout.doPaymentsDetailsCall.mock.calls).toMatchSnapshot();
  });

  it('should return to checkout when signatures mismatch', () => {
      req.querystring.payload = 'mocked_payload_result';
      req.querystring.signature = 'mismatching_signature';
      const URLUtils = require('dw/web/URLUtils');
      const adyenHelper = require('*/cartridge/scripts/util/adyenHelper');
      adyenHelper.getOrderMainPaymentInstrumentType.mockReturnValue('AdyenComponent');
      showConfirmation(req, res, jest.fn());
      expect(URLUtils.url.mock.calls[0][0]).toEqual('Error-ErrorCode');
  })

  it('should not continue processing when order is not open or failed', () => {
    const URLUtils = require('dw/web/URLUtils');
    req.querystring.merchantReference = 4;
    const adyenHelper = require('*/cartridge/scripts/util/adyenHelper');
    adyenHelper.getOrderMainPaymentInstrumentType.mockReturnValue('AdyenComponent');
    showConfirmation(req, res, jest.fn());
    expect(URLUtils.url.mock.calls[0][0]).toEqual('Cart-Show');
  })

  test.each(['Authorised', 'Pending', 'Received'])(
    'should handle successful payment: %p for SFRA6',
    (a) => {
      const adyenCheckout = require('*/cartridge/scripts/adyenCheckout');
      const adyenHelper = require('*/cartridge/scripts/util/adyenHelper');
      adyenHelper.getOrderMainPaymentInstrumentType.mockReturnValue('AdyenComponent');
      adyenConfigs.getAdyenSFRA6Compatibility.mockReturnValue(true);
      adyenCheckout.doPaymentsDetailsCall.mockImplementation(() => ({
        resultCode: a,
        paymentMethod: [],
        merchantReference: 'mocked_merchantReference',
      }));
      req.querystring.redirectResult = 'mocked_redirect_result';
      showConfirmation(req, res, jest.fn());
      expect(res.render.mock.calls[0][0]).toBe('orderConfirmForm');
    },
  );
  test.each(['Authorised', 'Pending', 'Received'])(
  'should handle successful payment: %p for SFRA5',
      (a) => {
        const adyenCheckout = require('*/cartridge/scripts/adyenCheckout');
        const adyenHelper = require('*/cartridge/scripts/util/adyenHelper');
        adyenHelper.getOrderMainPaymentInstrumentType.mockReturnValue('AdyenComponent');
        adyenConfigs.getAdyenSFRA6Compatibility.mockReturnValue(false);
        const URLUtils = require('dw/web/URLUtils');
        adyenCheckout.doPaymentsDetailsCall.mockImplementation(() => ({
          resultCode: a,
          paymentMethod: [],
          merchantReference: 'mocked_merchantReference',
        }));
        req.querystring.redirectResult = 'mocked_redirect_result';
        showConfirmation(req, res, jest.fn());
        expect(URLUtils.url.mock.calls[0][0]).toEqual('Order-Confirm');
      },
  );
});
