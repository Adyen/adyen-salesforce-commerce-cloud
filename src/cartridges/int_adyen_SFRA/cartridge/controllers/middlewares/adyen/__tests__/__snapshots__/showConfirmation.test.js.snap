// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show Confirmation should have payload 1`] = `Array []`;

exports[`Show Confirmation should have redirectResult 1`] = `Array []`;
